models/Import/Service/ResultImporter.php

```diff
@@ -356,4 +366,38 @@ private function getResultStorage(): AbstractRdsResultStorage
             )
         );
     }
+
+    /**
+     * Zero don't change SCORE_TOTAL value, but we want to update its timestamp when we have new value
+     * @param ImportResultInput $input
+     * @return bool
+     * @throws ImportResultException
+     * @throws common_exception_Error
+     */
+    private function checkOutcomeWithNewZeroValue(ImportResultInput $input): bool
+    {
+        $deliveryExecutionUri = $input->getDeliveryExecutionId();
+        $resultStorage = $this->getResultStorage();
+
+        foreach ($input->getOutcomes() as $itemId => $outcomes) {
+            $callItemId = $this->createCallItemId($deliveryExecutionUri, $itemId);
+            foreach ($outcomes as $outcomeId => $outcomeValue) {
+                $outcomeVariable = $this->getItemVariable(
+                    $resultStorage,
+                    $deliveryExecutionUri,
+                    $itemId,
+                    $callItemId,
+                    $outcomeId
+                );
+
+                /** @var taoResultServer_models_classes_Variable $variable */
+                $variable = $outcomeVariable['variable'];
+                if ($outcomeValue === 0.0 && $variable->getExternallyGraded() === false) {
```

Comment:
did not understand how this condition checks that a value is "new" for a variable